lib/filewatch/helper.rb

```diff
@@ -0,0 +1,72 @@
+# encoding: utf-8
+# code downloaded from Ruby on Rails 4.2.1
+# https://raw.githubusercontent.com/rails/rails/v4.2.1/activesupport/lib/active_support/core_ext/file/atomic.rb
+# change method name to avoid borking active_support and vice versa
+require 'fileutils'
+
+module FileHelper
+  extend self
+  # Write to a file atomically. Useful for situations where you don't
+  # want other processes or threads to see half-written files.
+  #
+  #   File.write_atomically('important.file') do |file|
+  #     file.write('hello')
+  #   end
+  #
+  # If your temp directory is not on the same filesystem as the file you're
+  # trying to write, you can provide a different temporary directory.
```

Comment:
should we add a check for this? or we could create the tmp file in the same directory as the target file?

Reply:
I don't know. Its copied verbatim from the rails code. We control the call to this method.

Reply:
ok, upon closer inspection the implementation is not sync'ed with the comments. there is no usage of a temp directory and this implementation does not match the rails link above.